### ISSUE TITLE
Terminals left in zombie state after quit and restart

### ISSUE THREAD

**Issue by gtritchie** [rstudio/rstudio#5871]
### System details

    RStudio Edition : Desktop or Server (macOS dev)
    RStudio Version : 1.3.659
    OS Version      : macOS 10.15.1
    R Version       : 3.6.0

### Steps to reproduce the problem
- start Desktop IDE
- start several terminals, say 5
- Cmd+Q to quit
- restart IDE
- click Terminal Pane

### Describe the problem in detail
Terminals are in the zombie state (where the output buffer is shown, but no process is attached), ending with various non-zero exit codes:
```
[Process completed]
[Exit code: 15]
```

### Describe the behavior you expected
Terminals should restart when viewed and reload their buffers.

Believe caused by 3ec8f86963b8d33bd075230fd2140ac6a0e493f8

Also, on Dev Server, I'm seeing instances of this error being logged:
```
11 Dec 2019 00:10:58 [rsession-gary] ERROR jsonrpc error 8 (Parameter missing) [description: missing default value for preference 'terminal_auto_close']; OCCURRED AT T rstudio::session::prefs::Preferences::readPref(const std::string &) [T = bool] src/cpp/session/prefs/UserPrefValues.cpp:68; LOGGED FROM: T rstudio::session::prefs::Preferences::readPref(const std::string &) [T = bool] src/cpp/session/prefs/UserPrefValues.cpp:71
```

**Comment by gtritchie** [comment]
The error being logged about `terminal_auto_close` preference is triggered by this call:

https://github.com/rstudio/rstudio/blob/2e05f760751af302b1ada5345f85128e9dd2996d/src/cpp/session/SessionConsoleProcess.cpp#L654

**Comment by gtritchie** [comment]
Fixed by the changes for #5872
